music_assistant/helpers/podcast_parsers.py

```diff
@@ -16,6 +19,28 @@
 )
 
 
+async def get_podcastparser_dict(
+    *, session: aiohttp.ClientSession, feed_url: str, max_episodes: int = 0
+) -> dict[str, Any]:
+    """Get feed parsed by podcastparser by providing the url.
+
+    max_episodes = 0 does not limit the returned episodes.
+    """
+    response: aiohttp.ClientResponse | None = None
+    # without user agent, some feeds can not be retrieved
+    # https://github.com/music-assistant/support/issues/3596
+    # but, reports on discord show, that also the opposite may be true
+    for headers in [{"User-Agent": "Mozilla/5.0"}, {}]:
+        response = await session.get(feed_url, headers=headers)
+        if response.status == 200:
+            break
+    if response is None or response.status != 200:
+        raise RuntimeError
```

Comment:
```suggestion
        response = await session.get(feed_url, headers=headers, raise_for_status=True)
```

So you can simply add raise_for_status as argument to do the status check for you